### ISSUE TITLE
Use port other than 80?

### ISSUE THREAD

**Issue by slashrsm** [edjafarov/remote-impress#3]
Hi, 

I'd like to thank you for this great tool first! It is awesome!! 

Is it possible to use remote-impress on a port different than 80? I have apache running on 80 and I'd like to run remote-impress on some other port. This would allow me to run presentation and remote on same computer. 

Is that possible?

JAnez

**Comment by edjafarov** [comment]
Sure. Just set up RMPORT env variable as you want.
What OS do you use?

Eldar Djafarov

sr. Web Developer in Luxoft

JS Maniac
http://eldar.djafarov.com
http://twitter.com/edjafarov

On Sun, Jun 3, 2012 at 10:02 PM, Janez Urevc <
reply@reply.github.com

**Comment by slashrsm** [comment]
Ubuntu 12.04. 

THX. I'll try that.

**Comment by slashrsm** [comment]
Hm... Does not help. I still get:

   info  - socket.io started
   warn  - error raised: Error: listen EADDRINUSE

**Comment by edjafarov** [comment]
are you setting it up like 
$ export RMPORT=3000 ???
if it is not working try to check env variables. It should.
Contact me if it's not:)

**Comment by slashrsm** [comment]
Just like this. Will this set the remote's port or webserver's port?

I am currently running apache on 80 and I'd like to run remote on 8081.

**Comment by edjafarov** [comment]
It should set up both.
Giver me few hours I will make it configurable from commandline. I.e.
remote port=8081 host=yourHostName
Eldar Djafarov

sr. Web Developer in Luxoft

JS Maniac
http://eldar.djafarov.com
http://twitter.com/edjafarov

On Mon, Jun 4, 2012 at 9:26 AM, Janez Urevc <
reply@reply.github.com

**Comment by slashrsm** [comment]
Great. Thanks!

**Comment by slashrsm** [comment]
Now I managed to start remote-impress on port 8899. It starts fine and displays it's welcome page. However, when I add 

```
<script src="http://localhost:8899/socket.io/socket.io.js"></script>
<script src="http://localhost:8899/remote.js"></script>
```

loading of remote fails. It looks like this happens because presentation tries to load remote from port 80

```
http://localhost/socket.io/1/?t=1338793733046
Failed to load resource: the server responded with a status of 404 (Not Found)
```

**Comment by slashrsm** [comment]
I took a look at the code and I managed to make it work. My solution is hardcoded, but it still points out where is the problem.

``` patch
diff --git a/remote.js b/remote.js
index 91a5282..8c0adc0 100755
--- a/remote.js
+++ b/remote.js
@@ -7,7 +7,7 @@ var events = require('events').EventEmitter;
 var ejs = require('ejs');
 var fs = require('fs');
 var observer = new events();
-var host = process.env.RMHOST;
+var host = process.env.RMHOST + ':8899';
 var remotes = {};
 var getIp = require('./getIP.js').getIP;
```

I am not fammiliar with node.js at all, so excuse me if I totally missed the point.

**Comment by edjafarov** [comment]
I agree that is stupid. I need to change the code:) and write some more docs.
although you can change RMHOST variable to yourhostname:8899 rather than yourhostname instead of hardcoding it.

**Comment by slashrsm** [comment]
Oh, yes. I am so stupid :)

No problem. It is a great tool. I will use it a lot.

Would it be possible to control more slideshows at once? Presenter would open it's presentation and give people link, that would attach their presentations to the same remote. That would be really useful for sessions beeing streamed over internet.
